File: Java/Problem_1614/Problem_1614.java
// Problem - 1614: Maximum Nesting Depth of the Parentheses
// Java Solution!
class Solution {
    public int maxDepth(String s) {
        int res = 0, curDepth = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(')
                curDepth++;
            else if (s.charAt(i) == ')')
                curDepth--;

            res = Math.max(curDepth, res);
        }

        return res;
    }
}

public static void main(String[] args) {
    Solution sol = new Solution();

    System.out.println(sol.maxDepth("(1+(2*3)+((8)/4))+1"));
    System.out.println(sol.maxDepth("(1)+((2))+(((3)))"));
}

File: Kotlin/Problem_1614.kt
import Kotlin.math.*

# Problem - 1614: Maximum Nesting Depth of the Parentheses
# Kotlin Solution!
class Solution {
    fun maxDepth(s: String): Int {
        var res = 0;
        var curDepth = 0;
        
        for (i in s){
            if (i == '(')
                curDepth++
            else if (i == ')')
                curDepth--;

            res = max(res, curDepth);
        }

        return res;
    }
}

fun main(){
    var sol = Solution();

    println(sol.maxDepth("(1+(2*3)+((8)/4))+1"));
    println(sol.maxDepth("(1)+((2))+(((3)))"));
}